internal/downloader/download.go

```diff
@@ -135,7 +168,31 @@ func (d *Downloder) download(desc, urlToGet, destination string, mode os.FileMod
 			fmt.Fprintln(os.Stderr, " done.")
 		}),
 	)
+	hasher := sha256.New()
+
+	_, err = io.Copy(io.MultiWriter(f, bar, hasher), resp.Body)
+	if err != nil {
+		return fmt.Errorf(
+			"Error while downloading text of %s into file %s: %v",
+			urlToGet, temporaryDestinationFile.Name(), err)
+	}
 
-	_, err = io.Copy(io.MultiWriter(f, bar), resp.Body)
-	return err
+	shaURLToGet := urlToGet + ".sha256"
+	shaExpected, err := d.getContentsOfURL(shaURLToGet)
```

Comment:
I would switch the download order: first download the shasum, then download the actual file.

Reply:
Done in b7f28cd